packages/intent-filler/src/core/filler.ts

```diff
@@ -0,0 +1,244 @@
+		this.pendingOrders.clear()
+
+		// Clear recheck counts
+		this.orderRecheckCount.clear()
+
+		// Wait for all queues to complete
+		const promises = []
+		this.chainQueues.forEach((queue) => {
+			promises.push(queue.onIdle())
+		})
+		promises.push(this.globalQueue.onIdle())
+
+		Promise.all(promises).then(() => {
+			console.log("All orders processed, filler stopped")
+		})
+	}
+
+	// Operations
+
+	private handleNewOrder(order: Order): void {
+		// Use the global queue for the initial analysis
+		// This can happen in parallel for PublicClient orders
+		this.globalQueue.add(async () => {
+			try {
+				// Check if the order has enough confirmations
+				const hasEnoughConfirmations = await this.checkConfirmations(order)
+
+				if (!hasEnoughConfirmations) {
+					// If not enough confirmations, add to pending queue with a delay
+					this.addToPendingQueue(order)
+					return
+				}
+
+				// If we have enough confirmations, proceed with strategy evaluation
+				this.evaluateAndExecuteOrder(order)
+			} catch (error) {
+				console.error(`Error processing order ${order.id}:`, error)
+			}
+		})
+	}
+
+	private async checkConfirmations(order: Order): Promise<boolean> {
+		try {
+			const sourceClient = this.chainClientManager.getPublicClient(order.sourceChain)
+			const orderValue = await this.calculateOrderValue(order, sourceClient)
+			const requiredConfirmations = this.config.confirmationPolicy.getConfirmationBlocks(
+				chainIds[order.sourceChain as keyof typeof chainIds],
+				orderValue,
+			)
+
+			const sourceReceipt = await sourceClient.getTransactionReceipt({ hash: order.transactionHash! })
+			const sourceConfirmations = await sourceClient.getTransactionConfirmations({
+				transactionReceipt: sourceReceipt,
+			})
+
+			console.log("sourceConfirmations", sourceConfirmations)
+			console.log("requiredConfirmations", requiredConfirmations)
+
+			if (sourceConfirmations < requiredConfirmations) {
+				console.debug(
+					`Insufficient confirmations for order ${order.id}, ${sourceConfirmations} confirmations, ${requiredConfirmations} required`,
+				)
+				return false
+			}
+
+			return true
+		} catch (error) {
+			console.error(`Error checking confirmations for order ${order.id}:`, error)
+			return false
+		}
+	}
+
+	private async calculateOrderValue(order: Order, client: PublicClient): Promise<bigint> {
+		let totalUSDValue = BigInt(0)
+
+		for (const input of order.inputs) {
+			const decimals = await this.contractService.getTokenDecimals(input.token, order.sourceChain)
+			const tokenUsdPrice = await fetchTokenUsdPriceOnchain(
+				input.token == bytes20ToBytes32(ADDRESS_ZERO)
+					? this.configService.getWrappedNativeAssetWithDecimals(order.sourceChain).asset
+					: input.token,
+				decimals,
+			)
+
+			totalUSDValue = totalUSDValue + BigInt(input.amount * BigInt(tokenUsdPrice))
+		}
+
+		return totalUSDValue
+	}
+
+	private addToPendingQueue(order: Order): void {
+		if (this.pendingOrders.has(order.id!)) {
```

Comment:
Why not use setInterval and cleart the interval once the number of confirmations is reached instead of having to do these checks?